File: Java/src/main/java/Solution901_1000/Solution942.java
package Solution901_1000;

/**
 * @author windows
 */
public class Solution942 {
    public int[] diStringMatch(String s) {
        int[] res = new int[s.length() + 1];
        int min = 0;
        int max = s.length();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'D'){
                res[i] = max--;
            }else{
                res[i] = min++;
            }
        }
        res[s.length()] = min;
        return res;
    }


    public static void main(String[] args) {
        Solution942 solution942 = new Solution942();
    }
}

File: C++/src/Solution901_1000/Solution942.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution942.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> diStringMatch(string s) {
  int n = s.size();
  vector<int> res;
  int left = 0, right = n;
  for (int i = 0; i < n; ++i) {
    res.push_back(s[i] == 'I' ? left++ : right--);
  }
  res.push_back(left);
  return res;
}
